system/ee/legacy/database/DB_active_rec.php

```diff
@@ -302,15 +303,21 @@ public function join($table, $cond, $type = '')
         $this->_track_aliases($table);
 
         // Strip apart the condition and protect the identifiers
-        if (preg_match('/([\w\.]+)([\W\s]+)(.+)/', $cond, $match)) {
+        if (is_empty($alias) && preg_match('/([\w\.]+)([\W\s]+)(.+)/', $cond, $match)) {
```

Comment:
I think this line is causing the tests to fail.  This probably just needs to change from `is_empty($alias)` to `empty($alias)`

Reply:
@bryannielsen updated